config-template.xml

```diff
@@ -17,9 +17,11 @@
     <engine name="browser" spec="5.0.3" />
     <plugin name="cordova-plugin-whitelist" spec="1.3.3" />
     <plugin name="cordova-sqlite-storage" spec="2.2.1" />
+    <plugin name="cordova-plugin-file" spec="6.0.1" />
     <plugin name="phonegap-plugin-barcodescanner" spec="7.0.2" />
     <plugin name="cordova-android-support-gradle-release" spec="1.2.0">
         <variable name="ANDROID_SUPPORT_VERSION" value="26.+" />
     </plugin>
     <plugin name="cordova-plugin-inappbrowser" spec="2.0.2" />
+    <preference name="AndroidExtraFilesystems" value="files,files-external,documents,sdcard,cache,cache-external,assets,root" />
```

Comment:
Are all these required? For example, is "root" really needed?

Reply:
Looks like root is required here.... I don't think Android provides a way to read/write to a shared/public directory - instead of granting access to the whole filesystem via the root directive.